tests/reporting/test_market_unit.py

```diff
@@ -451,9 +439,94 @@ def test_market_migrate_through_one_fork(localFixture, initializeMarket, constan
     assert initializeMarket.getLastReportersDisputeBond() == longToHexString(0)
     assert initializeMarket.getTentativeWinningPayoutDistributionHash() == initializeMarket.getDesignatedReportPayoutHash()
 
+def test_market_disavow_tokens(localFixture, initializeMarket, mockUniverse, mockReportingWindow, mockMarket, mockStakeTokenFactory):
+    stakePayNumeration = [0,0,0,0,numTicks]
+    with raises(TransactionFailed, message="market state has to be AWAITING_FORK_MIGRATION"):
+        initializeMarket.disavowTokens()
+
+    mockUniverse.setForkingMarket(mockMarket.address)
+    mockReportingWindow.setIsForkingMarketFinalized(False)
+    with raises(TransactionFailed, message="market needs stake tokens"):
+        initializeMarket.disavowTokens()
```

Comment:
I think this is actually a contract bug.  If there are no stake tokens, disavowing should just be a noop.  cc @nuevoalex

Reply:
code on contract is 
`require(stakeTokens.getCount() > 0);`

Reply:
Yeah, that feels wrong.  It feels like it should be `if (stakeTokens.getCount == 0) return true;`

Reply:
I'll make that change and submit as a separate PR.

Reply:
I'm fine with that change